tests/app/forms/test_questionnaire_form.py

```diff
@@ -97,7 +101,233 @@ def test_date_range_to_precedes_from_raises_question_error(self):
 
             form.validate()
             self.assertEqual(form.data, expected_form_data)
-            self.assertEqual(form.question_errors['reporting-period-question'], schema.error_messages['INVALID_DATE_RANGE'], AnswerStore())
+            self.assertEqual(form.question_errors['reporting-period-question'], schema.error_messages
+                             ['INVALID_DATE_RANGE'], AnswerStore())
+
+    def test_invalid_calculation_type(self):
+        store = AnswerStore()
+
+        answer_total = Answer(
+            answer_id='total-answer',
+            answer_instance=1,
+            group_instance=1,
+            value=10,
+        )
+
+        store.add(answer_total)
+
+        with self.test_request_context():
+            schema = load_schema_from_params('test', 'sum_equal_validation_against_total')
+
+            block_json = schema.get_block('breakdown-block')
+
+            question_json = {
+                'id': 'breakdown-question',
+                'type': 'Calculated',
+                'calculated': {
+                    'calculated_type': 'subtraction',
+                    'answer_id': 'total-answer',
+                    'answers_to_calculate': [
+                        'breakdown-1',
+                        'breakdown-2'
+                    ],
+                    'conditions': [
+                        'equals'
+                    ]
+                },
+                'answers': [{
+                    'id': 'breakdown-1',
+                    'label': 'Breakdown 1',
+                    'type': 'Number'
+                }, {
+                    'id': 'breakdown-2',
+                    'label': 'Breakdown 2',
+                    'type': 'Number'
+                }]
+            }
+
+            data = {
+                'breakdown-1': '3',
+                'breakdown-2': '5',
+            }
+
+            form = generate_form(schema, block_json, data, store)
+
+            with self.assertRaises(Exception):
+                with patch('app.questionnaire.questionnaire_schema.QuestionnaireSchema.get_questions_for_block',
+                           return_value=[question_json]):
+                    form.validate()
```

Comment:
This shouldn't just check the raising of an exception. It should also check it's of the . correct type with the correct message.